integrations/whatsapp/src/misc/whatsapp.ts

```diff
@@ -0,0 +1,161 @@
+import { IntegrationContext, Request, z } from '@botpress/sdk'
+import axios from 'axios'
+import * as bp from '.botpress'
+
+export class MetaOauthClient {
+  private clientId: string
+  private clientSecret: string
+  private version: string = 'v19.0'
+
+  constructor(private logger: bp.Logger) {
+    this.clientId = bp.secrets.CLIENT_ID
+    this.clientSecret = bp.secrets.CLIENT_SECRET
+  }
+
+  async getAccessToken(code: string) {
+    const query = new URLSearchParams({
+      client_id: this.clientId,
+      client_secret: this.clientSecret,
+      redirect_uri: `${process.env.BP_WEBHOOK_URL}/oauth`,
+      code,
+    })
+
+    const res = await axios.get(`https://graph.facebook.com/${this.version}/oauth/access_token?${query.toString()}`)
+    const data = z
+      .object({
+        access_token: z.string(),
+      })
+      .parse(res.data)
+
+    return data.access_token
+  }
+
+  async getWhatsappBusinessIdFromToken(input_token: string) {
```

Comment:
nitpick: the methods of this class have a mixture of camelCase and snake_case for the parameter names, could we make sure they're all camelCase just for consistency?

```suggestion
  async getWhatsappBusinessIdFromToken(inputToken: string) {
```